### ISSUE TITLE
feat(typeahead): Handling array of objects containing  value/label pairs

### ISSUE THREAD

**Issue by marco-gagliardi** [valor-software/ngx-bootstrap#2335]
With the `typeaheadOptionField` input it is possibile to pick a specific property to be used when the source is an array of objects. But what if I wanted to show one of the object's property as the input value, but assign another object's property as the model value? 

Say the source is something like this:
```
[
    {id: 1, name: "Alabama"},
    {id: 2, name: "Alaska"},
    {id: 3, name: "Arizona"}
]
```

It is not possibile to make the user see the `name` property in the form control, and bind the model  to the `id` property instead. Something like..

```
<input [(ngModel)]="customSelected"
       [typeahead]="statesComplex"
       typeaheadBindTo="id"
       typeaheadOptionField ="name"
       class="form-control">
```

.. or even better to the entire object itself (sometimes very useful):

```
<input [(ngModel)]="customSelected"   state
       [typeahead]="let state of statesComplex"
       typeaheadBindTo="state"
       typeaheadOptionField ="state.name"
       class="form-control">
```

**Comment by theCrius** [comment]
This has been suggested nearly one year ago and still is not supported.

**Comment by exhtml** [comment]
+1

**Comment by Domainv** [comment]
Closed as duplicate of #749
